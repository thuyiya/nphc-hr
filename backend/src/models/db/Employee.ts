const mongoose = require("mongoose");

const employeeSchema = new mongoose.Schema(
  {
    full_name: {
      required: true,
      type: String,
    },
    login_id: {
      required: true,
      type: String,
    },
    salary: {
      required: true,
      type: Number,
    },
    gender: {
      required: true,
      type: String,
      enum: ['Male', 'Female', 'Unknown'],
      default: "Unknown"
    },
    age: {
      type: Number,
      default: 20
    },
    activities: {
      type: Array,
    },
    profile_pic: {
      required: false,
      type: String,
      default: 'https://img.icons8.com/cute-clipart/2x/gender-neutral-user.png'
    },
  },
  {
    timestamps: {
      createdAt: "created_at",
      updatedAt: "updated_at",
    },
  }
);

export default mongoose.model('Employee', employeeSchema);
