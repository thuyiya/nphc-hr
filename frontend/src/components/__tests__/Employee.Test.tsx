import renderer from 'react-test-renderer';
import Employee from "../Employee";

describe('<Employee />', () => {
    it('Snapshot', () => {
        const component = renderer.create(<Employee />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})