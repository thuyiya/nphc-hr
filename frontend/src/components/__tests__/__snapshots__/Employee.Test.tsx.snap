// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div
    className="ant-row"
    style={Object {}}
  >
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="ant-space ant-space-horizontal ant-space-align-center input-container"
        style={Object {}}
      >
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <div
            className="input-prefix-icon"
          >
            <span
              aria-label="file-search"
              className="anticon anticon-file-search"
              role="img"
              style={
                Object {
                  "fontSize": 24,
                }
              }
            >
              <svg
                aria-hidden="true"
                data-icon="file-search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M688 312v-48c0-4.4-3.6-8-8-8H296c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8zm-392 88c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H296zm144 452H208V148h560v344c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V108c0-17.7-14.3-32-32-32H168c-17.7 0-32 14.3-32 32v784c0 17.7 14.3 32 32 32h272c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm445.7 51.5l-93.3-93.3C814.7 780.7 828 743.9 828 704c0-97.2-78.8-176-176-176s-176 78.8-176 176 78.8 176 176 176c35.8 0 69-10.7 96.8-29l94.7 94.7c1.6 1.6 3.6 2.3 5.6 2.3s4.1-.8 5.6-2.3l31-31a7.9 7.9 0 000-11.2zM652 816c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <div
            className="filter-input-prefix"
          >
            <span
              className="ant-typography filter-input-text-title"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Minimum Salary
            </span>
            <span
              className="ant-typography ant-typography-secondary filter-input-text-description"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Enter Amount
            </span>
          </div>
        </div>
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <span
            className="ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-affix-wrapper-borderless"
            onMouseDown={[Function]}
          >
            <span
              className="ant-input-prefix"
            >
              <span
                aria-label="dollar"
                className="anticon anticon-dollar"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="dollar"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm47.7-395.2l-25.4-5.9V348.6c38 5.2 61.5 29 65.5 58.2.5 4 3.9 6.9 7.9 6.9h44.9c4.7 0 8.4-4.1 8-8.8-6.1-62.3-57.4-102.3-125.9-109.2V263c0-4.4-3.6-8-8-8h-28.1c-4.4 0-8 3.6-8 8v33c-70.8 6.9-126.2 46-126.2 119 0 67.6 49.8 100.2 102.1 112.7l24.7 6.3v142.7c-44.2-5.9-69-29.5-74.1-61.3-.6-3.8-4-6.6-7.9-6.6H363c-4.7 0-8.4 4-8 8.7 4.5 55 46.2 105.6 135.2 112.1V761c0 4.4 3.6 8 8 8h28.4c4.4 0 8-3.6 8-8.1l-.2-31.7c78.3-6.9 134.3-48.8 134.3-124-.1-69.4-44.2-100.4-109-116.4zm-68.6-16.2c-5.6-1.6-10.3-3.1-15-5-33.8-12.2-49.5-31.9-49.5-57.3 0-36.3 27.5-57 64.5-61.7v124zM534.3 677V543.3c3.1.9 5.9 1.6 8.8 2.2 47.3 14.4 63.2 34.4 63.2 65.1 0 39.1-29.4 62.6-72 66.4z"
                  />
                </svg>
              </span>
            </span>
            <input
              className="ant-input ant-input-lg ant-input-borderless"
              hidden={null}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Amount Ex. 1000"
              style={null}
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "margin": 16,
          }
        }
      >
        -
      </div>
    </div>
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="ant-space ant-space-horizontal ant-space-align-center input-container"
        style={Object {}}
      >
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <div
            className="filter-input-prefix"
          >
            <span
              className="ant-typography filter-input-text-title"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Maximum Salary
            </span>
            <span
              className="ant-typography ant-typography-secondary filter-input-text-description"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Enter Amount
            </span>
          </div>
        </div>
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <span
            className="ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-affix-wrapper-borderless"
            onMouseDown={[Function]}
          >
            <span
              className="ant-input-prefix"
            >
              <span
                aria-label="dollar"
                className="anticon anticon-dollar"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="dollar"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm47.7-395.2l-25.4-5.9V348.6c38 5.2 61.5 29 65.5 58.2.5 4 3.9 6.9 7.9 6.9h44.9c4.7 0 8.4-4.1 8-8.8-6.1-62.3-57.4-102.3-125.9-109.2V263c0-4.4-3.6-8-8-8h-28.1c-4.4 0-8 3.6-8 8v33c-70.8 6.9-126.2 46-126.2 119 0 67.6 49.8 100.2 102.1 112.7l24.7 6.3v142.7c-44.2-5.9-69-29.5-74.1-61.3-.6-3.8-4-6.6-7.9-6.6H363c-4.7 0-8.4 4-8 8.7 4.5 55 46.2 105.6 135.2 112.1V761c0 4.4 3.6 8 8 8h28.4c4.4 0 8-3.6 8-8.1l-.2-31.7c78.3-6.9 134.3-48.8 134.3-124-.1-69.4-44.2-100.4-109-116.4zm-68.6-16.2c-5.6-1.6-10.3-3.1-15-5-33.8-12.2-49.5-31.9-49.5-57.3 0-36.3 27.5-57 64.5-61.7v124zM534.3 677V543.3c3.1.9 5.9 1.6 8.8 2.2 47.3 14.4 63.2 34.4 63.2 65.1 0 39.1-29.4 62.6-72 66.4z"
                  />
                </svg>
              </span>
            </span>
            <input
              className="ant-input ant-input-lg ant-input-borderless"
              hidden={null}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Amount Ex. 1000"
              style={null}
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <button
          className="ant-btn ant-btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="upload"
            className="anticon anticon-upload"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="upload"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
          <span>
            Upload Emplyee
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="ant-row"
    style={Object {}}
  >
    <h4
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
          "marginBottom": 16,
          "marginTop": 16,
        }
      }
    >
      Employees
    </h4>
    <div
      className="ant-col ant-col-24"
      style={Object {}}
    >
      <div
        className="ant-table-wrapper"
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-table ant-table-empty"
            >
              <div
                className="ant-table-container"
              >
                <div
                  className="ant-table-content"
                  onScroll={[Function]}
                  style={Object {}}
                >
                  <table
                    style={
                      Object {
                        "tableLayout": "auto",
                      }
                    }
                  >
                    <colgroup />
                    <thead
                      className="ant-table-thead"
                    >
                      <tr>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Id
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Name
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Login
                        </th>
                        <th
                          className="ant-table-cell ant-table-column-has-sorters"
                          colSpan={null}
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                          tabIndex={0}
                        >
                          <div
                            className="ant-table-column-sorters"
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                          >
                            <span
                              className="ant-table-column-title"
                            >
                              Salary
                            </span>
                            <span
                              className="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                className="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  className="anticon anticon-caret-up ant-table-column-sorter-up"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  className="anticon anticon-caret-down ant-table-column-sorter-down"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Action
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      className="ant-table-tbody"
                    >
                      <tr
                        className="ant-table-placeholder"
                        style={
                          Object {
                            "display": null,
                          }
                        }
                      >
                        <td
                          className="ant-table-cell"
                          colSpan={5}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          <div
                            className="ant-empty ant-empty-normal"
                          >
                            <div
                              className="ant-empty-image"
                            >
                              <svg
                                className="ant-empty-img-simple"
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <g
                                  fill="none"
                                  fillRule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    className="ant-empty-img-simple-ellipse"
                                    cx="32"
                                    cy="33"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    className="ant-empty-img-simple-g"
                                    fillRule="nonzero"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      className="ant-empty-img-simple-path"
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <div
                              className="ant-empty-description"
                            >
                              No Data
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Employee /> Snapshot 1`] = `
<div>
  <div
    className="ant-row"
    style={Object {}}
  >
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="ant-space ant-space-horizontal ant-space-align-center input-container"
        style={Object {}}
      >
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <div
            className="input-prefix-icon"
          >
            <span
              aria-label="file-search"
              className="anticon anticon-file-search"
              role="img"
              style={
                Object {
                  "fontSize": 24,
                }
              }
            >
              <svg
                aria-hidden="true"
                data-icon="file-search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M688 312v-48c0-4.4-3.6-8-8-8H296c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8zm-392 88c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H296zm144 452H208V148h560v344c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V108c0-17.7-14.3-32-32-32H168c-17.7 0-32 14.3-32 32v784c0 17.7 14.3 32 32 32h272c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm445.7 51.5l-93.3-93.3C814.7 780.7 828 743.9 828 704c0-97.2-78.8-176-176-176s-176 78.8-176 176 78.8 176 176 176c35.8 0 69-10.7 96.8-29l94.7 94.7c1.6 1.6 3.6 2.3 5.6 2.3s4.1-.8 5.6-2.3l31-31a7.9 7.9 0 000-11.2zM652 816c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <div
            className="filter-input-prefix"
          >
            <span
              className="ant-typography filter-input-text-title"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Minimum Salary
            </span>
            <span
              className="ant-typography ant-typography-secondary filter-input-text-description"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Enter Amount
            </span>
          </div>
        </div>
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <span
            className="ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-affix-wrapper-borderless"
            onMouseDown={[Function]}
          >
            <span
              className="ant-input-prefix"
            >
              <span
                aria-label="dollar"
                className="anticon anticon-dollar"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="dollar"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm47.7-395.2l-25.4-5.9V348.6c38 5.2 61.5 29 65.5 58.2.5 4 3.9 6.9 7.9 6.9h44.9c4.7 0 8.4-4.1 8-8.8-6.1-62.3-57.4-102.3-125.9-109.2V263c0-4.4-3.6-8-8-8h-28.1c-4.4 0-8 3.6-8 8v33c-70.8 6.9-126.2 46-126.2 119 0 67.6 49.8 100.2 102.1 112.7l24.7 6.3v142.7c-44.2-5.9-69-29.5-74.1-61.3-.6-3.8-4-6.6-7.9-6.6H363c-4.7 0-8.4 4-8 8.7 4.5 55 46.2 105.6 135.2 112.1V761c0 4.4 3.6 8 8 8h28.4c4.4 0 8-3.6 8-8.1l-.2-31.7c78.3-6.9 134.3-48.8 134.3-124-.1-69.4-44.2-100.4-109-116.4zm-68.6-16.2c-5.6-1.6-10.3-3.1-15-5-33.8-12.2-49.5-31.9-49.5-57.3 0-36.3 27.5-57 64.5-61.7v124zM534.3 677V543.3c3.1.9 5.9 1.6 8.8 2.2 47.3 14.4 63.2 34.4 63.2 65.1 0 39.1-29.4 62.6-72 66.4z"
                  />
                </svg>
              </span>
            </span>
            <input
              className="ant-input ant-input-lg ant-input-borderless"
              hidden={null}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Amount Ex. 1000"
              style={null}
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "margin": 16,
          }
        }
      >
        -
      </div>
    </div>
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="ant-space ant-space-horizontal ant-space-align-center input-container"
        style={Object {}}
      >
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <div
            className="filter-input-prefix"
          >
            <span
              className="ant-typography filter-input-text-title"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Maximum Salary
            </span>
            <span
              className="ant-typography ant-typography-secondary filter-input-text-description"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Enter Amount
            </span>
          </div>
        </div>
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <span
            className="ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-affix-wrapper-borderless"
            onMouseDown={[Function]}
          >
            <span
              className="ant-input-prefix"
            >
              <span
                aria-label="dollar"
                className="anticon anticon-dollar"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="dollar"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm47.7-395.2l-25.4-5.9V348.6c38 5.2 61.5 29 65.5 58.2.5 4 3.9 6.9 7.9 6.9h44.9c4.7 0 8.4-4.1 8-8.8-6.1-62.3-57.4-102.3-125.9-109.2V263c0-4.4-3.6-8-8-8h-28.1c-4.4 0-8 3.6-8 8v33c-70.8 6.9-126.2 46-126.2 119 0 67.6 49.8 100.2 102.1 112.7l24.7 6.3v142.7c-44.2-5.9-69-29.5-74.1-61.3-.6-3.8-4-6.6-7.9-6.6H363c-4.7 0-8.4 4-8 8.7 4.5 55 46.2 105.6 135.2 112.1V761c0 4.4 3.6 8 8 8h28.4c4.4 0 8-3.6 8-8.1l-.2-31.7c78.3-6.9 134.3-48.8 134.3-124-.1-69.4-44.2-100.4-109-116.4zm-68.6-16.2c-5.6-1.6-10.3-3.1-15-5-33.8-12.2-49.5-31.9-49.5-57.3 0-36.3 27.5-57 64.5-61.7v124zM534.3 677V543.3c3.1.9 5.9 1.6 8.8 2.2 47.3 14.4 63.2 34.4 63.2 65.1 0 39.1-29.4 62.6-72 66.4z"
                  />
                </svg>
              </span>
            </span>
            <input
              className="ant-input ant-input-lg ant-input-borderless"
              hidden={null}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Amount Ex. 1000"
              style={null}
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <button
          className="ant-btn ant-btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="upload"
            className="anticon anticon-upload"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="upload"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
          <span>
            Upload Emplyee
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="ant-row"
    style={Object {}}
  >
    <h4
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
          "marginBottom": 16,
          "marginTop": 16,
        }
      }
    >
      Employees
    </h4>
    <div
      className="ant-col ant-col-24"
      style={Object {}}
    >
      <div
        className="ant-table-wrapper"
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-table ant-table-empty"
            >
              <div
                className="ant-table-container"
              >
                <div
                  className="ant-table-content"
                  onScroll={[Function]}
                  style={Object {}}
                >
                  <table
                    style={
                      Object {
                        "tableLayout": "auto",
                      }
                    }
                  >
                    <colgroup />
                    <thead
                      className="ant-table-thead"
                    >
                      <tr>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Id
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Name
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Login
                        </th>
                        <th
                          className="ant-table-cell ant-table-column-has-sorters"
                          colSpan={null}
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                          tabIndex={0}
                        >
                          <div
                            className="ant-table-column-sorters"
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                          >
                            <span
                              className="ant-table-column-title"
                            >
                              Salary
                            </span>
                            <span
                              className="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                className="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  className="anticon anticon-caret-up ant-table-column-sorter-up"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  className="anticon anticon-caret-down ant-table-column-sorter-down"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Action
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      className="ant-table-tbody"
                    >
                      <tr
                        className="ant-table-placeholder"
                        style={
                          Object {
                            "display": null,
                          }
                        }
                      >
                        <td
                          className="ant-table-cell"
                          colSpan={5}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={Object {}}
                        >
                          <div
                            className="ant-empty ant-empty-normal"
                          >
                            <div
                              className="ant-empty-image"
                            >
                              <svg
                                className="ant-empty-img-simple"
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <g
                                  fill="none"
                                  fillRule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    className="ant-empty-img-simple-ellipse"
                                    cx="32"
                                    cy="33"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    className="ant-empty-img-simple-g"
                                    fillRule="nonzero"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      className="ant-empty-img-simple-path"
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <div
                              className="ant-empty-description"
                            >
                              No Data
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
